packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -585,6 +516,96 @@ bool validateJavaGradle(Logger logger,
   return false;
 }
 
+// Returns compatibility information for the valid range of Gradle versions for
+// the specified Java version.
+JavaGradleCompat? getValidGradleVersionRangeForJavaVersion(
+  Logger logger, {
+  required String javaV,
+}) {
+  for (final JavaGradleCompat data in _javaGradleCompatList) {
+    if (isWithinVersionRange(javaV, min: data.javaMin, max: data.javaMax, inclusiveMax: false)) {
+      return data;
+    }
+  }
+
+  logger.printTrace('Unable to determine valid Gradle version range for Java version $javaV.');
+  return null;
+}
+
+// Validate that the specified Java and Android Gradle Plugin (AGP) versions are
```

Comment:
I think these should be /// for public dart doc. Please mention what null inputs return. Might want to mention that true means definitely compatible whereas false is the assumed default.